dig/README.md

```diff
@@ -0,0 +1,78 @@
+# Dependency Injection Graph
+
+This package provides a fairly opinionated way of resolving object dependencies.
+There are two sides of dig: `Inject` and `Resolve`
+
+## Inject
+
+Adding an object, or a constructor of an object to the graph.
+
+There are two ways of providing an object:
+
+1. Injecting a pointer to an existing object
+1. Injecting a "constructor function" that returns one pointer (or interface)
+
+### Injecting an object
```

Comment:
Nit: Injecting --> Inject

Verb+ing as a heading can be good for people skimming, but "Inject" is more direct. Plus, you have "Injecting an object" in the paragraph that follows for anyone searching for those terms instead.